### PR TITLE
[14.0][MIG]product_state

### PR DESCRIPTION
**Pull request by watthanun**

### CODE REVIEW

File: product_state/tests/test_product_state.py [hunk 1]
```diff
@@ -6,16 +8,8 @@ class TestProductState(SavepointCase):
...
-        )
+        cls.state = cls.ProductState.create({"name": "State Name", "code": "Code"})
 
     def test_01_product_state(self):
-        print(self.state.products_count)
-
-    
-
-
+        # print(self.state.products_count)
```
Comment (rousseldenis) [inline product_state/tests/test_product_state.py]: Please remove the comments and logger

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

**Review by andreampiovesana — APPROVED** [review]
go

**Review by andreampiovesana — APPROVED** [review]
i am using in production, i changed manifest 
    "application": False,
go

**Comment by andreampiovesana** [comment]
you should chang manifest
"application": False,

**Comment by bosd** [comment]
@watthanun Are you going to finish this one? Please look at the commits so V14 changes are in the V14 migration commit.

**Review by ferranmateu — APPROVED** [review]
Functional review OK but better quit it from Applications

**Comment by rousseldenis** [comment]
@watthanun @kittiu Do you plan to finish this one ?

**Comment by rousseldenis** [comment]
@watthanun @kittiu I did a PR on your repo to fix this and add some tests https://github.com/ecosoft-odoo/product-attribute/pull/2

**Comment by kittiu** [comment]
> @watthanun @kittiu I did a PR on your repo to fix this and add some tests [ecosoft-odoo#2](https://github.com/ecosoft-odoo/product-attribute/pull/2)

Ok. Really sorry. I will merge yours.

**Comment by rousseldenis** [comment]
> > @watthanun @kittiu I did a PR on your repo to fix this and add some tests [ecosoft-odoo#2](https://github.com/ecosoft-odoo/product-attribute/pull/2)
> 
> Ok. Really sorry. I will merge yours.

No problem, this way is good too. I think I will add another one too for some improvements (tracking and one index)

**Comment by rousseldenis** [comment]
@Cedric-Pigeon @emagdalenaC2i I think the state should be required if the module is installed.

The values for existing products are initialized through the init_hook and there is a default now.

What do you think ?

**Review by emagdalenaC2i — APPROVED** [review]

**Comment by rousseldenis** [comment]
@kittiu Could you fix pre-commit, so we can go forward on this ?

**Comment by kittiu** [comment]
> @kittiu Could you fix pre-commit, so we can go forward on this ?

Done, thanks!

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**